### PR TITLE
Tl/runtime features

### PR DESCRIPTION
**Pull request by Torpheus**

### CODE REVIEW

File: metamorph-clib/src/dyntypes/dyntypes.c [hunk 1]
```diff
@@ -85,7 +88,8 @@ void release_dyntype(dyntype_t dyntype){
         case(SCHEME_TYPE_STRING): {
             scheme_string_t string;
             string = *dyntype.data.string_val;
-            //release_string(string);
+            if (dyntype._mutable)
```
Comment (rapgru) [inline metamorph-clib/src/dyntypes/dyntypes.c]: do not use `_mutable` flag of `dyntype_t` for marking strings created via c literal strings. this flag is for the scheme notion of mutable storage locations only.

File: metamorph-clib/src/functions/activations.c [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+//Does not release previous activation, because previous_activation is a weak reference
+void gc_activation(activation_t* activation) {
+    //Never release activation when in computation or in ongoing release 
+    if (activation->n_computations || !activation->previous_activation)
+        return;
+
+    //Release activation only if all references to it are cycle refernces
+    if (activation->n_captures) {
+        //int ref = 0;
```
Comment (rapgru) [inline metamorph-clib/src/functions/activations.c]: is this necessary?

File: metamorph-clib/src/functions/functions.c [hunk 3]
```diff
@@ -2,302 +2,73 @@
 #include "../functions.h"
 #include "../continuation.h"
 #include "../lambda.h"
+#include "../activations.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (rapgru) [inline metamorph-clib/src/functions/functions.c]: remove unnecessary include

**Review by rapgru — CHANGES_REQUESTED** [review]
delete test_functions.i

File: metamorph-clib/src/functions/activations.c (hunk 2 above)
Comment (Torpheus) [reply metamorph-clib/src/functions/activations.c]: yeees

**Review by rapgru — APPROVED** [review]

**Pull request merged**